packages/thirdweb/src/react/web/ui/ConnectWallet/Modal/ConnectModalContent.tsx

```diff
@@ -97,23 +95,10 @@
         setScreen(reservedScreens.getStarted);
       }}
       selectWallet={(newWallet) => {
-        setScreen(newWallet);
-
-        // When selecting Coinbase, fire off the connection immediately
-        // this fixes an issue on safari where the CB popup window would get blocked
-        // because of the delay between this click and the actual connection
-        // that happens within 'CoinbaseSDKConnection' component 2 layers down
-        // when the cb extension is installed -> pops the extension
-        // when the extension is not installed -> pops a new window
-        // then in 'CoinbaseSDKConnection' the connection is requested again
-        // which just "links" to the extension / popup window a few ms later
-        // and handle the completion / error handling there
-        // this connection flow structure needs to be refactored to properly handle this
-        if (newWallet.id === COINBASE) {
-          newWallet.connect({
-            client,
-            chain,
-          });
+        if (newWallet.onConnectRequested) {
+          newWallet.onConnectRequested().then(() => setScreen(newWallet));
```

Comment:
can this fail? if so what do we do if it fails?

Reply:
let me add a catch to be safe